### DISCUSSION TITLE
Hardware acceleration for Odroid C4

### DISCUSSION THREAD

**Discussion by chris669** [blakeblackshear/frigate#6230] [Ask A Question]
Sorry couldn't find any informations about hardware acceleration parameters for my Odroid C4, here are the specifications, if someone could help!  At this time I do not use Coral, will in the future. Now testing with just 1 cam. Thanks in advance.

Amlogic S905X3 12nm Processor
L1 instruction cache: 32 KB, 4-way set associative (128 sets), 64 byte lines, shared by 1 processor

L1 data cache: 32 KB, 4-way set associative (128 sets), 64 byte lines, shared by 1 processor

L3 data cache: 512KB , 16-way set associative (512 sets), 64 byte lines, shared by 4 processors

Quad-Core Cortex-A55 (2.016GHz)

ARMv8-A architecture with Neon and Crypto extensions

Mali-G31 MP2 GPU with 4 x Execution Engines (650Mhz)

**Comment by NickM-27** [comment]
I don't think it's currently supported.

**Comment by RichardPar** [comment]
> Sorry couldn't find any informations about hardware acceleration parameters for my Odroid C4, here are the specifications, if someone could help! At this time I do not use Coral, will in the future. Now testing with just 1 cam. Thanks in advance.
> 
> Amlogic S905X3 12nm Processor L1 instruction cache: 32 KB, 4-way set associative (128 sets), 64 byte lines, shared by 1 processor
> 
> L1 data cache: 32 KB, 4-way set associative (128 sets), 64 byte lines, shared by 1 processor
> 
> L3 data cache: 512KB , 16-way set associative (512 sets), 64 byte lines, shared by 4 processors
> 
> Quad-Core Cortex-A55 (2.016GHz)
> 
> ARMv8-A architecture with Neon and Crypto extensions
> 
> Mali-G31 MP2 GPU with 4 x Execution Engines (650Mhz)

To test if Video is accelerated.. run
```
ffmpeg -c:v h264_v4l2m2m -i rtsp://localhost:8554/<camera_feed_here> -f null /dev/null
```
replace the -i section with what you have (even if its just a file) 
paste the output of the playback. Look at the line

```frame=   39 fps= 24 q=-0.0 Lsize=N/A time=00:00:03.08 bitrate=N/A speed=1.87x```
Frame should increment
  

Additionally, login to another window and see the CPU utilisation.

(run this on the platform - regardless of docker)

**Reply by chris669** [reply]
thanks will try this and keep you informed.
Need to install addon like ssh and web terminal to run this, right?

**Reply by RichardPar** [reply]
you can run this on the platform natively - 

to give it a try.. add the following to your frigate.yml

```
ffmpeg:
   global_args: -hide_banner -loglevel debug
   input_args: preset-rtsp-generic
   hwaccel_args:
     - -c:v:1
     - h264_v4l2m2m   
   output_args:
     detect: -threads 4 -f rawvideo
```

notice the -c:v:1   (the '1' on the end output format .. you can start with -c:v  )

**Reply by jpduyx** [reply]
should this be done on a global level or if you have a few different camera's you need to set the hwaccel_args according to the stream of each camera  -c:v:1 vs -c:v:0 or something ?
